src/main/kotlin/net/lumalyte/lg/infrastructure/persistence/migrations/SQLiteMigrations.kt

```
@@ -1383,4 +1413,112 @@ class SQLiteMigrations(private val plugin: JavaPlugin, private val connection: C
+     * v20: per-home rank whitelist + per-ally-guild inbound ally-home whitelist + USE_ALLY_HOMES permission backfill.
+     *
+     * Adds `guild_homes.allowed_ranks` (TEXT, CSV of rank UUIDs) and
+     * `guilds.ally_home_allowed_guilds` (TEXT, CSV of guild UUIDs).
+     * Backfills existing homes with all current ranks of the owning guild (policy B,
+     * see 2026-05-10-rank-and-home-perms-design.md §2.3), and adds USE_ALLY_HOMES
+     * to every existing rank's permissions (§3.1).
+     *
+     * Idempotent: ALTER TABLE wrapped in try/catch for "duplicate column", and
+     * backfill UPDATE only writes rows where the new columns are NULL.
+     */
+    private fun migrateToVersion20() {
```

detekt (1.23.8)

[warning] The function migrateToVersion20 appears to be too complex based on Cognitive Complexity (complexity: 35). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)